src/otx/backend/native/engine.py

```diff
@@ -825,7 +826,7 @@ def from_config(
             data=datamodule,
             model=model,
             **engine_kwargs,
-        )
+        ), train_kwargs
```

Comment:
I'm not sure it's better to return kwargs from `from_engine`. Could we keep it cleaner by simply passing `max_epochs` directly to `OTXEngine.from_config(...)` to override it, instead of returning and modifying the returned kwargs and then passing them to `engine.train`? I think the workflow would be smoother if we override parameters within `from_config` itself, rather than handling it externally.

Reply:
This is the problem. In `from_config` we add max_epoch to enigine_kwargs and store them in the cache.
But in train method, we override it by passing number of epochs from train method. Argument in train method has priority. To avoid this, I propose to return `train_kwargs`. I didn't find a solution how we can override it directly in `OTXEngine.from_config(...)`.

We return `train_kwargs` similarly in ConfigConverter.

Reply:
Do we really need so much complexity for the train function?
And also,
"current_value == param.default" - user may want to  specifically use default parameter, but it will be overridden with self._train_overrides.

Reply:
I find it a bit awkward that `from_config` returns a `train_kwargs` dictionary, which then requires the user to manually override values before passing them to `engine.train`.

For example:
Option 1:
```py
engine, train_kwargs = OTXEngine.from_config(...)
train_kwargs["max_epochs"] = epoch
engine.train(**train_kwargs)
```
vs.

Option 2:
```py
engine = OTXEngine.from_config(..., max_epochs=epoch)
engine.train()
```
I think Option 2 looks cleaner and more intuitive, as it avoids the extra indirection and keeps the override logic within `from_config`.